lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -0,0 +1,612 @@
+                current_disk['disk_unit_number'] = disk['unit_number']
+
+            else:
+                self.module.fail_json(msg="Please specify 'unit_number' under disk parameter"
+                                          " at index [%s], which is required while creating disk." % disk_index)
+
+            # Type of Disk
+            disk_type = disk.get('type', 'thick').lower()
+            if disk_type not in ['thin', 'thick', 'eagerzeroedthick']:
+                self.module.fail_json(msg="Invalid 'disk_type' specified for disk index [%s]. Please specify"
+                                          " 'disk_type' value from ['thin', 'thick', 'eagerzeroedthick']." % disk_index)
+            current_disk['disk_type'] = disk_type
+
+            # SCSI Controller Type
+            scsi_contrl_type = disk.get('scsi_type', 'paravirtual').lower()
+            if scsi_contrl_type not in self.scsi_device_type.keys():
+                self.module.fail_json(msg="Invalid 'scsi_type' specified for disk index [%s]. Please specify"
+                                          " 'scsi_type' value from ['%s']" % (disk_index,
+                                                                              "', '".join(self.scsi_device_type.keys())))
+            current_disk['scsi_type'] = scsi_contrl_type
+
+            disks_data.append(current_disk)
+        return disks_data
+
+    def get_recommended_datastore(self, datastore_cluster_obj=None):
+        """
+        Function to return Storage DRS recommended datastore from datastore cluster
+        Args:
+            datastore_cluster_obj: datastore cluster managed object
+
+        Returns: Name of recommended datastore from the given datastore cluster,
+                 Returns None if no datastore recommendation found.
+
+        """
+        # Check if Datastore Cluster provided by user is SDRS ready
+        sdrs_status = datastore_cluster_obj.podStorageDrsEntry.storageDrsConfig.podConfig.enabled
+        if sdrs_status:
+            # We can get storage recommendation only if SDRS is enabled on given datastorage cluster
+            pod_sel_spec = vim.storageDrs.PodSelectionSpec()
+            pod_sel_spec.storagePod = datastore_cluster_obj
+            storage_spec = vim.storageDrs.StoragePlacementSpec()
+            storage_spec.podSelectionSpec = pod_sel_spec
+            storage_spec.type = 'create'
+
+            try:
+                rec = self.content.storageResourceManager.RecommendDatastores(storageSpec=storage_spec)
+                rec_action = rec.recommendations[0].action[0]
+                return rec_action.destination.name
+            except Exception as e:
+                # There is some error so we fall back to general workflow
+                pass
+        datastore = None
+        datastore_freespace = 0
+        for ds in datastore_cluster_obj.childEntity:
+            if isinstance(ds, vim.Datastore) and ds.summary.freeSpace > datastore_freespace:
```

Comment:
I may be mistaken but, I think isinstance(ds, vim.Datastore) is not necessary since you get the datastores list from get_all_objs(self.content, [vim.Datastore]) (so all ds should be of type vim.Datastore).

@Akasurde you tell me "OK", but forget to make the change.